pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -2156,18 +2158,36 @@ _PrimReaderContext::AddOutOfSchemaProperty(
     // Get the converter and add the property.
     const bool isOutOfSchema = true;
     const UsdAbc_AlembicType alembicType(*header);
-    const SdfValueTypeName usdTypeName =
+    SdfValueTypeName usdTypeName =
         _context.GetSchema().GetConversions().FindConverter(alembicType);
+
     if (usdTypeName) {
-        _PrimReaderContext::Property &prop = 
-            (TfGetEnvSetting(USD_ABC_WRITE_UV_AS_ST_TEXCOORD2FARRAY) && 
-             name == UsdAbcPropertyNames->uvIndices)? 
-                (_AddProperty(UsdAbcPropertyNames->stIndices, 
-                             usdTypeName, header->getMetaData(), 
-                             sampleTimes, isOutOfSchema)) :
-                (_AddProperty(TfToken(name), 
-                             usdTypeName, header->getMetaData(), 
-                             sampleTimes, isOutOfSchema)); 
+        TfToken propName(name);
+        if (TfGetEnvSetting(USD_ABC_WRITE_UV_AS_ST_TEXCOORD2FARRAY) &&
+            name == UsdAbcPropertyNames->uvIndices) {
+            propName = UsdAbcPropertyNames->stIndices;
+        }
+        else if (TfStringEndsWith(name, _tokens->indicesSuffix)){
+            // If the property is indexed and we're adding the indices here
+            // then we want to make sure they are added as an IntArray
+            // as that's the only supported type for index arrays.
+            usdTypeName = SdfValueTypeNames->IntArray;
+        }
+        else if (TfStringEndsWith(name, _tokens->valuesSuffix)) {
+            // If the property is indexed and we're adding the values
+            // it might have a ":vals" suffix. We need to remove it
+            // to be compatible.
+            propName = TfToken(name.substr(0, name.size() - _tokens->valuesSuffix.size()));
```

Comment:
In the comment you indicate that the ":vals" suffix may or may not be there... this calculation will break if they are not there.  More robust to do a find() of the suffix if the comment is accurate.

Reply:
Apologies - brain malfunction; you’ve of course already determined it has the suffix.